tools/actioninjector.v2/internal/cmd/random.go

```diff
@@ -0,0 +1,296 @@
+	rawExecutionGasPrice int64
+	executionGasPrice    *big.Int
+	rawExecutionData     string
+	executionData        []byte
+	retryNum             uint64
+	retryInterval        time.Duration
+	duration             time.Duration
+	resetInterval        time.Duration
+	aps                  int
+	workers              uint64
+}{}
+
+func inject(args []string) string {
+	var err error
+	injectCfg.transferPayload, err = hex.DecodeString(injectCfg.rawTransferPayload)
+	if err != nil {
+		return fmt.Sprintf("failed to decode payload %s: %v.", injectCfg.transferPayload, err)
+	}
+	injectCfg.executionData, err = hex.DecodeString(injectCfg.rawExecutionData)
+	if err != nil {
+		return fmt.Sprintf("failed to decode data %s: %v", injectCfg.rawExecutionData, err)
+	}
+	injectCfg.transferGasPrice = big.NewInt(injectCfg.rawTransferGasPrice)
+	injectCfg.executionGasPrice = big.NewInt(injectCfg.rawExecutionGasPrice)
+	injectCfg.executionAmount = big.NewInt(injectCfg.rawExecutionAmount)
+	p, err := newInjectionProcessor()
+	if err != nil {
+		return fmt.Sprintf("failed to create injector processor: %v.", err)
+	}
+
+	ctx, cancel := context.WithTimeout(context.Background(), injectCfg.duration)
+	defer cancel()
+	go p.injectProcess(ctx)
+	go p.resetProcess(ctx)
+	<-ctx.Done()
+	return ""
+}
+
+func init() {
+	flag := injectCmd.Flags()
+	flag.StringVar(&injectCfg.configPath, "injector-config-path", "./tools/actioninjector/gentsfaddrs.yaml", "path of config file of genesis transfer addresses")
+	flag.StringVar(&injectCfg.serverAddr, "addr", "127.0.0.1:14004", "target ip:port for grpc connection")
+	flag.Uint64Var(&injectCfg.transferGasLimit, "transfer-gas-limit", 20000, "transfer gas limit")
+	flag.Int64Var(&injectCfg.rawTransferGasPrice, "transfer-gas-price", 10, "transfer gas price")
+	flag.StringVar(&injectCfg.rawTransferPayload, "transfer-payload", "", "transfer payload")
+	flag.StringVar(&injectCfg.contract, "contract", "io1pmjhyksxmz2xpxn2qmz4gx9qq2kn2gdr8un4xq", "smart contract address")
+	flag.Int64Var(&injectCfg.rawExecutionAmount, "execution-amount", 50, "execution amount")
+	flag.Uint64Var(&injectCfg.executionGasLimit, "execution-gas-limit", 20000, "execution gas limit")
+	flag.Int64Var(&injectCfg.rawExecutionGasPrice, "execution-gas-price", 10, "execution gas price")
+	flag.StringVar(&injectCfg.rawExecutionData, "execution-data", "2885ad2c", "execution data")
+	flag.Uint64Var(&injectCfg.retryNum, "retry-num", 5, "maximum number of rpc retries")
+	flag.DurationVar(&injectCfg.retryInterval, "retry-interval", 1*time.Second, "sleep interval between two consecutive rpc retries")
+	flag.DurationVar(&injectCfg.duration, "duration", 60*time.Hour, "duration when the injection will run")
+	flag.DurationVar(&injectCfg.resetInterval, "reset-interval", 10*time.Second, "time interval to reset nonce counter")
+	flag.IntVar(&injectCfg.aps, "aps", 30, "actions to be injected per second")
+	flag.Uint64Var(&injectCfg.workers, "workers", 10, "number of workers")
```

Comment:
What are workers and how do they work?

Reply:
this is the total number of concurrent go routines the processor start with. it doesn't impact tps, since processor will add more if it doesn't have enough